lib/doorkeeper/config.rb

```diff
@@ -70,8 +70,26 @@ def scopes_by_grant_type(hash = {})
       # `params` object.
       #
       # @param methods [Array] Define client credentials
+      # @deprecated
       def client_credentials(*methods)
-        @config.instance_variable_set(:@client_credentials_methods, methods)
+        deprecated("client_credentials", "Use the client_authentication option instead. Automatically converting to client_authentication")
+
+        client_authentication = methods.map {|method|
+          case method
+          when :from_basic
+            :client_secret_basic 
+          when :from_params
+            :client_secret_post
+          else
+            Kernel.warn("[DOORKEEPER] Unknown client_credentials method detected: #{method}")
+          end
+        }.reject(&:nil?)
+
+        if client_authentication.empty?
+          Kernel.warn("[DOORKEEPER] No known client_credentials method detected, ignoring option")
+        else
+          @config.instance_variable_set(:@client_credentials_methods, client_authentication.concat([:none]))
+        end
       end
```

Comment:
This is pretty messy, but we basically have to handle all the different permutations this supported.

Reply:
This also now handles the custom authenticator case, where the `client_credentials` was a lambda or class.